jupyter_server/services/contents/manager.py

```diff
@@ -75,10 +76,30 @@ def emit(self, data):
 
     root_dir = Unicode("/", config=True)
 
+    preferred_dir = Unicode(
+        "/",
```

Comment:
I still find it odd to have a default value that is an absolute path when the trait calls for a _relative_ path. Is `"/"` in this context equivalent to `"."` from a path perspective?  This argument is exacerbated when seeing the default value for `root_dir`, which calls for an _absolute_ path, is also `"/"`.

(Actually, looking at the b/c handling above, we _temporarily_ set `preferred_dir` to `None`, then return `"/"` - so this also leads to the same oddity.  I suppose if we always treat it as a relative path and _always_ append its value to `root_dir` (while never testing for it being an absolute path - except for the drive letter stuff for Windows), we're okay as-is.)

I believe in the b/c validation handling when `preferred_dir` has the same value as `root_dir`, the `preferred_dir` value is set to `None`.  Should we add `allowNone = True`, and let the default value be `None` here?

Reply:
I see your point. Reading through the code, an API path seems to be treated like it *may* have a leading `/`, but that most internal use of it is always without it (`.strip('/')` is often called to remove both leading and trailing slashes). If so, the default should be changed to an empty string `""`. Does that sound better to you? If so, I guess we should also normalize it during the validator (i.e. call `.strip("/")`).

Reply:
Yes, an empty string makes sense; more so than `None` or `"."`. Thanks.